src/librustc_trait_selection/traits/error_reporting/mod.rs

```diff
@@ -283,6 +283,11 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             .unwrap_or(false);
                         let is_from = format!("{}", trait_ref.print_only_trait_path())
                             .starts_with("std::convert::From<");
+                        let is_unsize = {
```

Comment:
you can test whether something is the unsize trait by using the "lang items" -- you can use [`tcx.lang_items()`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.lang_items) to access them and then invoke [`unsize_trait`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/lang_items/struct.LanguageItems.html#method.unsize_trait). Then compare if `Some(trait_ref.def_id) == tcx.lang_Items().unsize_trait()`.

Reply:
I hadn't spotted I was in an `InferCtxt` so I didn't realise I had lang items available to me. Thanks for catching this!